ksqldb-execution/src/main/java/io/confluent/ksql/execution/util/ComparisonUtil.java

```diff
@@ -81,6 +84,21 @@ private static boolean handleBoolean(final Type operator, final SqlType right) {
         && (operator == Type.EQUAL || operator == Type.NOT_EQUAL);
   }
 
+  private static boolean handleArray(final Type operator, final SqlType right) {
+    return right.baseType() == SqlBaseType.ARRAY
+        && (operator == Type.EQUAL || operator == Type.NOT_EQUAL);
+  }
+
+  private static boolean handleMap(final Type operator, final SqlType right) {
+    return right.baseType() == SqlBaseType.MAP
+        && (operator == Type.EQUAL || operator == Type.NOT_EQUAL);
+  }
+
+  private static boolean handleStruct(final Type operator, final SqlType right) {
+    return right.baseType() == SqlBaseType.STRUCT
+        && (operator == Type.EQUAL || operator == Type.NOT_EQUAL);
+  }
```

Comment:
These all need to handle `Type.IS_DISTINCT_FROM` as well.

This also suggests you're missing QTT test cases covering `IS NOT DISTINCT FROM` and `IS DISTINCT FROM` for these three complex types.